airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/report_streams/report_streams.py

```diff
@@ -108,7 +108,7 @@ class ReportStream(BasicAmazonAdsStream, ABC):
         # Check if the connector received an error: 'Report date is too far in the past. Reports are only available for 60 days.'
         # In theory, it does not have to get such an error because the connector correctly calculates the start date,
         # but from practice, we can still catch such errors from time to time.
-        (406, "Report date is too far in the past."),
+        (406, re.compile(r"^Report date is too far in the past\.")),
```

Comment:
@grubberr won't this silently make the connector skip the report? I think it would be a better UX to update the start_date to `today - 60 days` if it's too far in the past so we can generate the reports. What do you think?

Reply:
@girarda 

### Internal connector operation 

For internal operation connector has all the needed code:
1. Connector shifts the start date to try to avoid the "60 days" error taking into account each profile timezone.

2. But even after (1) connector still can catch "60 days" error for example:
  a. calculating `start date` for profile-1, profile-2 - shifting if needed
  b. processing profile-1 during some long time
  c. processing profile-2 got "60 days" error because (b) taken pretty long time and boundary already violated.
  
  For case (2) we can still catch "60 days" error and all we can do just ignore it and continue to process next day.

Currently, AmazonAds already has all the needed code to work with "60 days" problem, there was only a simple bug with checking "60 days" error which was fixed in this PR.

### Additional UX improvement

We can improve the UX experience for customers. We can generate an error on the "check" command if the `start date` that the customer entered will be more than 60 days old.

but I would recommend using:

```
today - ( 60 days + 1 days )
```

We need this additional day because there is no strict boundary
We can have one AmazonAds account with multiple profiles which operate in **different timezones**.

Reply:
@grubberr this makes sense, thanks for the explanation! Would you mind moving this comment in the PR description for posterity, this is very useful context

> today - ( 60 days + 1 days )
Agreed.